### PR TITLE
chore: update pr template

### PR DESCRIPTION
**Pull request by g1eny0ung**
Signed-off-by: Yue Yang <g1enyy0ung@gmail.com>

### What problem does this PR solve?

After the new PR template was updated, I found that there are still some things that need improvement.

### What's changed and how it works?

- Remove `> Problem Summary:` and `> What's Changed:` quotes, they have some repetition with the title.
- Update **Related changes**
  - `Chaos Dashboard` to `Dashboard UI` to prevent ambiguity.
  - Add recent two releases to avoid handwriting. (Now you just need to check the box)
- Add **DCO** section because almost all new contributors forget to sign the DCO...

### Related changes

- [ ] Need to update `chaos-mesh/website`
- [ ] Need to update `Dashboard UI`
- Need to **cheery-pick to release branches**
  - [ ] release-2.0
  - [ ] release-1.2

### Checklist

Tests

- [ ] Unit test
- [ ] E2E test
- [ ] No code
- [ ] Manual test (add steps below)

Side effects

- [ ] Breaking backward compatibility

### Release note 

```text
Please add a release note.

You can safely ignore this section if you don't think this PR needs a release note.
```

### DCO

If you find the DCO check fails, please run commands like below (Depends on the actual situations. For example, if the failed commit isn't the most recent) to fix it:

```shell
git commit --amend --signoff
git push --force
```

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -18,20 +18,18 @@ Please follow the Title Formats below when you open a new PR:
...
-> What's Changed:
-
 ### Related changes
 
 - [ ] Need to update `chaos-mesh/website`
-- [ ] Need to update `Chaos Dashboard`
-- [ ] Need to **cheery-pick to release branches**
+- [ ] Need to update `Dashboard UI`
+- Need to **cheery-pick to release branches**
+  - [ ] release-2.0
```
Comment (milasuperstar) [inline .github/pull_request_template.md:31]: Does that mean we have to constantly update the newest releases here?

Comment (g1eny0ung) [reply .github/pull_request_template.md:31]: Yep. But not too often, only modify it whenever we release the next.

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -18,20 +18,18 @@ Please follow the Title Formats below when you open a new PR:
...
+  - [ ] release-1.2
```
Comment (STRRL) [inline .github/pull_request_template.md:32]:
```suggestion
  - [ ] release-2.1
  - [ ] release-2.0
```

rest LGTM :heart:

**Comment by g1eny0ung** [comment]
@milasuperstar @STRRL PTAL again ❤️.

**Review by STRRL — APPROVED** [review]
LGTM

**Comment by milasuperstar** [comment]
LGTM

**Review by milasuperstar — APPROVED** [review]

**Pull request merged**